### DISCUSSION TITLE
Prompt directory feature (hopeful)

### DISCUSSION THREAD

**Discussion by valenciavex** [starship/starship#3279] [Ideas]
Is it possible to have a hybrid prompt where a part of it is the current shell prompt and rest is the starship prompt? I don't know if the implementation would allow something like that. Right now it substitutes the whole prompt.

I'm asking this because of a super useful feature of the `fish` shell where it collapses the directory to its first letter in the prompt when you are in a nested file structure. 

The `truncation` option in the `directory` module is super powerful and beautiful in starship, but I would also like to pair it with the directory name collapsing feature.

Does this warrant raising an issue ?
Thank you :)

<img width="535" alt="Screen Shot 2021-11-26 at 11 20 15 PM" src="https://user-images.githubusercontent.com/80008736/143667802-f86ce6fd-dd80-48f1-86bd-747ceeb2baf3.png">
<img width="535" alt="Screen Shot 2021-11-26 at 11 20 47 PM" src="https://user-images.githubusercontent.com/80008736/143667815-04986764-34ee-414f-a723-5608b8ce34db.png">

**Comment by andytom** [comment]
Starship supports fish style truncation in the [directory module](https://starship.rs/config/#directory) see the advanced options section. To enable it add the following to your starship config file

```toml
[directory]
fish_style_pwd_dir_length = 1
```

**Reply by valenciavex** [reply]
I should have gone over the advanced options, sorry about that. But this is awesome! Best prompt ever!! Thanks a ton!!!
